chain-adapter/src/main/kotlin/com/d3/chainadapter/dedup/hazelcast/HazelcastBlockProcessor.kt

```diff
@@ -0,0 +1,87 @@
+package com.d3.chainadapter.dedup.hazelcast
+
+import com.d3.chainadapter.dedup.BlockProcessor
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.map.IMap
+import iroha.protocol.BlockOuterClass.Block
+import mu.KLogging
+import java.io.Closeable
+import java.util.concurrent.TimeUnit
+
+/**
+ * Name of distributed map and name of the key in the map. Same name for simplicity
+ */
+const val PROCESSED_BLOCK_ACCESS_KEY = "processed-block-numbers"
+
+/**
+ * How long we wait to acquire exclusive lock. If timeout exceed, then exception thrown
+ */
+const val LOCK_AWAITING_TIMEOUT_SECONDS = 30L
+
+/**
+ * This implementation sends new block to RMQ only if it was not processed before by this or another
+ * instance of service. It allows to run multiple instances of the service. Deduplication provided
+ * by pessimistic lock on distributed map.
+ */
+class HazelcastBlockProcessor(private val hazelcastInstance: HazelcastInstance) : BlockProcessor, Closeable {
+
+    private val processedBlockNumber : IMap<String, Long> = hazelcastInstance.getMap(PROCESSED_BLOCK_ACCESS_KEY)
+
+    override fun onNewBlock(block: Block, publish: (block: Block) -> Unit) {
```

Comment:
it is named `action` in the other implementation and `publish` here, could you choose the same name please?

Reply:
Fixed